pkg/collector/corechecks/checkbase.go

```diff
@@ -172,10 +172,19 @@ func (c *CheckBase) Warnf(format string, params ...interface{}) error {
 // long-running checks (persisting after Run() exits)
 func (c *CheckBase) Stop() {}
 
-// Cancel does nothing by default, you need to implement it if
+// Cancel calls CommonCancel by default. Override it if
 // your check has background resources that need to be cleaned up
-// when the check is unscheduled.
-func (c *CheckBase) Cancel() {}
+// when the check is unscheduled. Make sure to call CommonCancel from
+// your override.
```

Comment:
> Make sure to call CommonCancel from your override

Note: It is possible to remove the call `CommonCancel` in the override.

```go
func (c *CheckBase) Cancel() {
     c.doCheckBase()
	aggregator.DestroySender(c.checkID)
}

func (c *CheckBase) doCheckBase() {}
```

As there is no [override keyword](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/override) in Go, renaming temporary `Cancel` can help spotting  all the overrides of `Cancel` in a derived struct.

Reply:
Yes, this limitation is pretty similar to the `CheckBase.CommonConfigure` call that has to be made from an overridden `Configure`. I chose to use the same pattern here for consistency.